vnext/Microsoft.ReactNative/Views/TextViewManager.cpp

```diff
@@ -72,40 +74,53 @@ class TextShadowNode final : public ShadowNodeBase {
     Super::AddView(child, index);
 
     if (auto run = static_cast<ShadowNodeBase &>(child).GetView().try_as<winrt::Run>()) {
-      if (m_ColorValue) {
-        AddHighlighter(m_ColorValue.value(), run.Text().size());
+      if (m_backgroundColor) {
+        AddHighlighter(m_backgroundColor, m_foregroundColor, run.Text().size());
       }
-
       m_prevCursorEnd += run.Text().size();
     } else if (auto span = static_cast<ShadowNodeBase &>(child).GetView().try_as<winrt::Span>()) {
-      AddNestedTextHighlighter(m_ColorValue, span, static_cast<VirtualTextShadowNode &>(child).m_highlightData);
+      AddNestedTextHighlighter(
+          m_backgroundColor, m_foregroundColor, span, static_cast<VirtualTextShadowNode &>(child).m_highlightData);
     }
   }
 
   void AddNestedTextHighlighter(
       const std::optional<winrt::Windows::UI::Color> &parentColor,
+      const std::optional<winrt::Windows::UI::Color> &parentForeColor,
       winrt::Span &span,
       VirtualTextShadowNode::HighlightData highData) {
-    if (!highData.color && parentColor) {
-      highData.color = parentColor;
+    if (!highData.backgroundColor && parentColor) {
+      highData.backgroundColor = parentColor;
+    }
+
+    if (!highData.foregroundColor && parentForeColor) {
+      highData.foregroundColor = parentForeColor;
     }
 
     for (const auto &el : span.Inlines()) {
       if (auto run = el.try_as<winrt::Run>()) {
-        if (highData.color) {
-          AddHighlighter(highData.color.value(), run.Text().size());
+        if (highData.backgroundColor) {
+          AddHighlighter(highData.backgroundColor, highData.foregroundColor, run.Text().size());
         }
 
         m_prevCursorEnd += run.Text().size();
       } else if (auto spanChild = el.try_as<winrt::Span>()) {
-        AddNestedTextHighlighter(highData.color, spanChild, highData.data[highData.spanIdx++]);
+        AddNestedTextHighlighter(
+            highData.backgroundColor, highData.foregroundColor, spanChild, highData.data[highData.spanIdx++]);
       }
     }
   }
 
-  void AddHighlighter(const winrt::Windows::UI::Color &color, size_t runSize) {
+  void AddHighlighter(
+      const std::optional<winrt::Windows::UI::Color> &backgroundColor,
+      const std::optional<winrt::Windows::UI::Color> &foregroundColor,
+      size_t runSize) {
     auto newHigh = winrt::TextHighlighter{};
-    newHigh.Background(react::uwp::SolidBrushFromColor(color));
+    newHigh.Background(react::uwp::SolidBrushFromColor(backgroundColor.value()));
```

Comment:
If foreground color is optional, but background color is required, could we wire parameter types to reflect that? I.e. only use the std::optional where std::nullopt would be a valid input. 

Having a parameter type that doesn't allow null values acts as an explicit invariant to the assumption the highlighted code makes, that backgroundColor always has a value.

Reply:
True, `AddHighlighter` only gets called when `backgroundColor` is present, but since it's not always, I store it as std::optional in the shadow node (original PR remark). And then I just passed it around as such.
Given this, would it still be better to change this method's param to plain Color from optional, but keep the member field as optional? What about `AddNestedTextHighlighter` - it might get called with a null backgroundColor.